src/sweeper/sweepers/duplicates.py

```diff
@@ -7,52 +7,54 @@
 
 import arcpy
 import re
+import os
 from xxhash import xxh64
 
-
 class DuplicateTest(object):
-    def __init__(self):
+    def __init__(self, workspace, table_name):
+        self.report = {}
+        self.workspace = workspace
+        self.table_name = table_name
 
 
-    def sweep(self, fc):
+    def sweep(self):
 
         digestDict = {}
-        duplicateDict = {}
-        dupePrintLst = []
+
         digTrim = re.compile(r'(\d+\.\d{2})(\d+)')
 
-        shps = ['Shape', 'SHAPE']
+        arcpy.env.workspace = self.workspace
 
-        fields = [f.name for f in arcpy.ListFields(fc) if f.name not in shps]
+        omitFlds = ['Shape', 'SHAPE', 'GLOBALID']
+        fields = [f.name for f in arcpy.ListFields(self.table_name) if f.name not in omitFlds]
         fields.append('SHAPE@WKT')
 
-        with arcpy.da.SearchCursor(fc, fields) as sCursor:
+        with arcpy.da.SearchCursor(self.table_name, fields) as sCursor:
             for row in sCursor:
                 shp = row[-1]
                 if shp != None:
                     coordTrim = digTrim.sub(r'\1', shp)
-                    hash = xxh64('{} {}'.format(row[1:], coordTrim))
+                    hash = xxh64('{} {}'.format(row[1:-1], coordTrim))
                     digest = hash.hexdigest()
                     if digest not in digestDict:
                         digestDict.setdefault(digest)
                     else:
-                        duplicates = duplicateDict.setdefault('Duplicate Features', [])
-                        duplicates.extend(row[0])
-                else:
-                    duplicateDict.extend(row[0])
+                        self.report[row[0]] = 'duplicate feature'
```

Comment:
does this assume that the object id is always the 0 index of the row?

Reply:
yep

Reply:
I don't think that is a safe assumption and should be modified.